src/components/application_manager/test/app_launch/app_launch_ctrl_test.cc

```diff
@@ -202,7 +202,7 @@ TEST_F(AppLaunchCtrlTest, StoredAppIsLaunchedAfterDeviceConnected) {
   app_launch::ApplicationDataPtr app_to_launch = GetTestAppData(0);
   MockAppPtr app = GetTestApp(0);
 
-  TestAsyncWaiter waiter;
+  std::shared_ptr<TestAsyncWaiter> waiter = std::make_shared<TestAsyncWaiter>();
```

Comment:
@VladSemenyuk I think it would be great to create a static function inside of `TestAsyncWaiter` class like:
```
std::shared_ptr<TestAsyncWaiter> createInstance() {
  return std::make_shared<TestAsyncWaiter>();
}
```
And make a constructor private. Then in unit tests you can do everything much simpler:
```
auto waiter = TestAsyncWaiter::createInstance();
```

From another side, this will also guarantee that another developers in the future will not use default-constructed instances of waiter, which causes instability in UT

Reply:
Done